src/lib/field.ts

```diff
@@ -518,6 +518,42 @@ class Field {
     return new Field(z);
   }
 
+  /**
+   * Bitwise AND gate on {@link Field} elements. Equivalent to the [bitwise AND `&` operator in JavaScript](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Bitwise_AND).
+   * An AND gate works by comparing two bits and returning `1` if both bits are `1`, and `0` otherwise.
+   *
+   * The `length` parameter lets you define how many bits should be compared. It defaults to the size of the field in bits ({@link Fp.sizeInBits}).
+   *
+   * **Note:** Specifying a larger `length` parameter adds additional constraints.
+   *
+   * *Note:* Both {@link Field} elements need to fit into `2^length - 1`, or the operation will fail.
+   * For example, for `length = 2` (2² = 4), `.and` will fail for any element that is larger than `> 3`.
+   *
+   * @example
+   * ```ts
+   * let a = Field(5);    // ... 000101
+   * let b = Field(3);    // ... 000011
+   *
+   * let c = a.and(b);    // ... 000001
+   * c.assertEquals(1);
+   * ```
+   */
+  and(
+    y: Field | bigint | number | string,
+    length: number = Fp.sizeInBits
```

Comment:
the XOR PR uses a default bit length of 32. I think either 32 or 255 kinda make sense as a default, but the choice should be the same for XOR and AND